### PR TITLE
[Windows] Fix EGL surface destruction race

### PR DESCRIPTION
**Pull request by loic-sharma**
This fixes the `WindowsTest.EngineCanTransitionToHeadless` flakiness reported by @matanlurey.

EGL surfaces can only be used by a single thread at a time. Concurrent operations are unsafe.

Previously, the EGL surface was destroyed on the platform thread. This was safe as this always happened after the engine was shutdown and the raster thread was stopped. However, in a multi-view world a view can be destroyed while the engine is running. There may be pending raster tasks that operate on the render surface. Thus, the EGL surfaces should be destroyed on the raster thread when it is available.

This bug was introduced by https://github.com/flutter/engine/pull/51681
Part of https://github.com/flutter/flutter/issues/142845

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by matanlurey — APPROVED** [review]
RSLGTM, mostly to unblock the tree.

The code looks good (like, it's readable and the names and docs are good), I just can't tell you if it's "correct" for the domain, so defer to you on whether to land as-is or wait for more reviews.

File: shell/platform/windows/flutter_windows_view.cc
Comment (yaakovschectman) [inline shell/platform/windows/flutter_windows_view.cc:95]: What does this `latch` accomplish? Does this prevent the engine from destructing before this task runs?

**Review by yaakovschectman — APPROVED** [review]
LGTM mod one question

File: shell/platform/windows/flutter_windows_view.cc
Comment (loic-sharma) [reply shell/platform/windows/flutter_windows_view.cc]: Good question. This runs in the `FlutterWindowsView` destructor, which owns the `std::unique<egl::WindowSurface>`. The latch forced the destructor to wait until the surface was destroyed.

A cleaner version would be to just move the surface's ownership to the raster thread. I've updated it to that insead!

**Pull request merged**
